src/xrpld/rpc/detail/RPCHelpers.cpp

```diff
@@ -249,8 +249,18 @@ getAccountObjects(
     if (!dir)
     {
         // it's possible the user had nftoken pages but no
-        // directory entries
-        return mlimit < limit;
+        // directory entries. If there's no nftoken page, we will
+        // give empty array for account_objects.
+        if (mlimit >= limit)
+            jvResult[jss::account_objects] = Json::arrayValue;
+
+        // non-zero dirIndex validity was checked in the caller function;
+        // by this point, it should be zero.
```

Comment:
as commented elsewhere, I would prefer if this function did not rely on the validation of `dirIndex` happening in the caller, but rather move that check into the body of `getAccountObjects`

Reply:
updated